ln-simln-jamming/src/reputation_interceptor.rs

```diff
@@ -178,15 +178,88 @@ where
         })
     }
 
-    /// Bootstraps the reputation of nodes in the interceptor network using the historical forwards provided.
-    pub async fn bootstrap_network(
-        &mut self,
-        bootstrap: &BoostrapRecords,
-        general_jammed: &[(u64, PublicKey)],
-    ) -> Result<(), BoxError> {
-        self.bootstrap_network_history(bootstrap).await?;
+    pub async fn new_from_snapshot(
```

Comment:
Tests to cover validation cases and a very basic snapshot being loaded + checking that we're adding the right values for the network.